### ISSUE TITLE
Unhygienic Display impl

### ISSUE THREAD

**Issue by pie-flavor** [shepmaster/snafu#202]
Given this code:
```rust
use snafu::Snafu;

#[derive(Debug, Snafu)]
pub enum Error {
    #[snafu(display("{}", f))]
    Whatever {
        f: i32,
    }
}
```
`cargo build` produces:
```
error[E0599]: no method named `write_fmt` found for type `&i32` in the current scope
```
which is caused by the name `f` conflicting with the name of the `&mut Formatter` in the `Display` impl.

**Comment by shepmaster** [comment]
Yuck. Thank you for the report! I'm going to see if there's some nice solution to prevent that for good. Otherwise, I'll rename it to something really hard collide with and push up a patch version soon-ish.

Out of my own morbid curiosity, why do you have a field called `f`?

**Comment by pie-flavor** [comment]
@shepmaster It was short for file.

**Comment by shepmaster** [comment]
Should be fixed on 0.6.1, thanks again for the report!
